sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,196 @@
+        self._log_file_path = rospkg.RosPack().get_path('sr_wear_logger') + "/" + self._hand_serial + "/"
+        self._log_file_name = "wear_data.yaml"
+        self._init_log()
+
+        if self._hand_serial != "":
+            rospy.Timer(rospy.Duration(2), self._save_data_localy)
+            rospy.Timer(rospy.Duration(10), self._upload_to_AWS)
+
+        rospy.Subscriber('/joint_states', JointState, self.callback)
+
+    def _extract_hand_data(self, msg):
+        hand_data = dict(zip(msg.name, msg.position))
+        for key in hand_data.keys():
+            if key.startswith("ra") or key.startswith("la"):
+                hand_data.pop(key, None)
+        return hand_data
+
+    def callback(self, msg):
+        hand_data = self._extract_hand_data(msg)
+        updates = dict.fromkeys(hand_data.keys(), 0)
+        if self._first_run:
+            self._previous_values = hand_data
+            self._first_run = False
+        for key, value in hand_data.items():
+            value_difference = abs(self._previous_values[key] - value)
+            if value_difference > self._threshold:
+                self._current_values[key] += round(value_difference, 5)
+        self._previous_values = hand_data
+
+    def _get_latest_file_name_from_AWS(self):
+
+        files = self.aws_manager.get_bucket_structure_with_prefix("shadowrobot.benchmarks", self._hand_serial)
+        if files is None:
+            return ""
+
+        files.sort(key=lambda x: x['LastModified'])
+        last_file_path = files[-1]['Key']
+        latest_file_name_from_AWS = last_file_path[last_file_path.find("/")+1:]
+
+        return latest_file_name_from_AWS
+
+    def _upload_to_AWS(self, event):
+        success = False
+
+        orignal_file = self._log_file_path + self._log_file_name
+        now = datetime.datetime.now()
+        dated_file = now.strftime("%Y-%m-%d-%H-%M-%S") + ".yaml"
+        shutil.copy(orignal_file, self._log_file_path + dated_file)
+        rospy.sleep(1)
+
+        while os.stat(self._log_file_path + dated_file).st_size == 0:
+            shutil.copy(orignal_file, self._log_file_path + dated_file)
+            rospy.sleep(0.5)
+
+        success = self.aws_manager.upload("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                          self._hand_serial, [dated_file])
+        rospy.sleep(1)
+
+        if os.path.exists(self._log_file_path + dated_file):
+            os.remove(self._log_file_path + dated_file)
+
+        return success
+
+    def _download_from_AWS(self):
+        success = False
+        success = self.aws_manager.download("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                            self._hand_serial, [self._get_latest_file_name_from_AWS()])
+        if success:
+            rospy.sleep(1)
+            latest_file_name_from_AWS = self._get_latest_file_name_from_AWS()
```

Comment:
so what happens if the latest file is "", we should probably raise exception and handle it? or we gonna copy to empty file?

Reply:
Changed to copy only is path exists. I prefer to have a "" string returned because this string will be also passed to the path for download_from_aws.